### PR TITLE
docs: fix misleading step in getting started routing

### LINKED ISSUE #40280: Docs state to update as follows, but does not mention what to do
# 📚 Docs or angular.io bug report

### Description

 
On the page [Adding Navigation](https://angular.io/start/start-routing), there is a section where we need to update the products by adding a productId. The statement reads, and I quote:
> Update the *ngFor directive to read as follows. This statement instructs Angular to iterate over the items in the products array and assigns each index in the array to the productId variable when iterating over the list.

But there is nothing below that actually does this. Was this intentional? If it was, maybe rephrase the sentence to mean that **you** need to do it.

### What's the affected URL?**
 
https://angular.io/start/start-routing

### Expected vs Actual Behavior**

I expected a code stub that succeded this statement, indicating that **this** is what you need to do, as a newbie :sweat_smile: .

## 📷Screenshot

Screenshot from 2020-12-30 19-15-01

### Anything else relevant?

If this is an easy fix, maybe I myself can try this out. Thanks.

### PR DESCRIPTION
**Pull request by kapunahelewong**
The [getting started routing step #4](https://angular.io/start/start-routing#associate-a-url-path-with-a-component) about updating the `*ngFor` is misleading because the user doesn't actually update it specifically, rather the update is made to the anchor tag, within the `*ngFor`. There is no change from the first step of the tutorial to the `*ngFor`. "As follows" is not followed by a code sample, instead the code sample is in the next step, which is confusing. This PR removes the reference to the `*ngFor` and moves the explanation into the next step, where it is more appropriate. The change goes from:

```
<div *ngFor="let product of products">
  <h3> <a [title]="product.name + ' details'">{{ product.name }}</a></h3>
</div>

```

to

```
<div *ngFor="let product of products">
  <h3> <a [title]="product.name + ' details'" [routerLink]="['/products', product.id]">{{ product.name }}</a></h3>
</div>
```

You don't edit `<div *ngFor="let product of products">` so there is no code snippet to pull for it.

Fixes https://github.com/angular/angular/issues/40280

### CODE REVIEW

**Review by aikithoughts — APPROVED** [review]
LGTM. Thanks for fixing this.

Reviewed-for: global-docs-approvers

**Comment by kapunahelewong** [comment]
Closing this in favor of https://github.com/angular/angular/pull/40197, which includes the same change.
